src/Actions/Selection.ts

```diff
@@ -138,6 +142,37 @@ export class ActionSelection {
         return Promise.resolve(true);
     }
 
+    @PrototypeReflect.metadata(SymbolMetadata.Action.isChange, true)
+    static byTextObject(args: {
+        textObject: TextObject
+    }): Thenable<boolean> {
+
+        const activeTextEditor = window.activeTextEditor;
+
+        if (! activeTextEditor) {
+            return Promise.resolve(false);
+        }
+
+        let ranges: Range[] = [];
+
+        activeTextEditor.selections.forEach(selection => {
+            const match = args.textObject.apply(selection.active);
```

Comment:
Things are a little wired when the active cursor is at the start of the selection.